bundles/http/Citrix/SAP-Ariba/Ariba_v2.js

```diff
@@ -1,399 +1,410 @@
-	}
-	return items
+    for (let i = 0; i < items.length; i++) {
+        items[i] = flatJsonObject(items[i], 'supplier', 'supp_')
+        items[i] = flatJsonObject(items[i], 'accountCategory', 'acc_cat_')
+        items[i] = flatJsonObject(items[i], 'description', 'desc_')
+        items[i] = flatJsonObject(items[i], 'desc_price', 'desc_pr_')
+    }
+    return items
 }
 
 function flatLineItemsAccountings(accountings) {
-	for (let i = 0; i < accountings.length; i++) {
-		accountings[i] = flatJsonObject(accountings[i], 'generalLedger', 'gl_')
-		accountings[i] = flatJsonObject(accountings[i], 'costCenter', 'cc_')
-		accountings[i] = flatJsonObject(accountings[i], 'amount', 'am_')
-	}
-	return accountings
+    for (let i = 0; i < accountings.length; i++) {
+        accountings[i] = flatJsonObject(accountings[i], 'generalLedger', 'gl_')
+        accountings[i] = flatJsonObject(accountings[i], 'costCenter', 'cc_')
+        accountings[i] = flatJsonObject(accountings[i], 'amount', 'am_')
+    }
+    return accountings
 }
 
 function flatRequisition(requisition) {
-	requisition = flatJsonObject(requisition, 'totalCost', 'tc_')
-	requisition = flatJsonObject(requisition, 'requester', 'rq_')
-	return requisition
+    requisition = flatJsonObject(requisition, 'totalCost', 'tc_')
+    requisition = flatJsonObject(requisition, 'requester', 'rq_')
+    return requisition
 }
 
 async function parseAndStoreApprovalRequests(sync, json, totalCost, groups) {
-	//this function is resolving approvers for requisitions and copy necessary informations to create notifications
-	let rqUserList = []
-	for (let appReqsr of json) {
-		for (let approvers of appReqsr.approvers) {
-			if (approvers.type != undefined) {
-				try {
-					approvers.approvalRequired = appReqsr.approvalRequired
-					approvers.approvalState = appReqsr.approvalState
-					approvers.reason = appReqsr.reason
-					approvers.state = appReqsr.state
-					approvers.totalCostAmount = totalCost.amount
-					approvers.totalCostCurrency = totalCost.currency
-					if (approvers.type == 'user') { //if approver is a user
-						approvers = flatJsonObject(approvers, 'user', '')
-						rqUserList.push(approvers)
-					}
-					if (approvers.type == 'group') { //if approver is a group
-						if (groups.ids.includes(approvers.id) == false) {
-							groups = await addIfMissingGroup(sync, approvers.id, groups)
-						}
-						rqUserList = rqUserList.concat(resolveGroupApproval(approvers, groups))
-					}
-				}
-				catch (e) {
-					console.log('Warning: unable to get all requisition data for user')
-				}
-			}
-		}
-	}
-	sync.dataStore.save('pendingApprovablesRequisitions', rqUserList) //saving approvers
-	return groups
+    //this function is resolving approvers for requisitions and copy necessary informations to create notifications
+    let rqUserList = []
+    for (let appReqsr of json) {
+        for (let approvers of appReqsr.approvers) {
+            if (approvers.type !== undefined) {
```

Comment:
`null` is allowed? what about other types? what about something like this: `if (['user', 'group'].includees(approvers.type)) {`

Reply:
fixed